CrashMessage/CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+# CrashMessage
+
+add_executable(CrashMessage
+               "main.cpp"
+)
+
+set_target_properties(CrashMessage PROPERTIES
+                      RUNTIME_OUTPUT_DIRECTORY ${NS_BINARY_DIR}/bin
+                      OUTPUT_NAME CrashMsg
+                      LINK_FLAGS "/MANIFEST:NO /DEBUG /SUBSYSTEM:WINDOWS /ENTRY:mainCRTStartup"
+)
```

Comment:
This adds a new exe which might be an issue, see https://github.com/R2Northstar/NorthstarLauncher/pull/477#issuecomment-1654548775

Reply:
Alright, reasoning for new exe is in: https://github.com/R2Northstar/NorthstarLauncher/pull/477#issuecomment-1654565064

copy/paste from @F1F7Y's comment:

> [Additional exe is for] separating the message box from the game process. This means the game no-longer stays open untill the message box is closed.
> 
> Imo this is better for user experience as the game can close independently of the user interacting with the message box freeing up both system resources and not being annoying with fullscreen grabbing focus / taking extra time to close in some circumstances

Reply:
Tbh, the freeing up resources part is not that important IMO, as they will be freed up when box is closed anyway, so in my opinion it doesn't make much difference.

Reply:
Again my main concern is that we basically cannot remove it again from someone's install and we have also been very adamant to keep the user install "clean" for the most part.

Reply:
Separate exe is the superior choice, for a number of reasons:
1. system resources
2. no potential for crash handler crashing and causing an infinite crash loop (which has happened before)
3. fullscreen with titanfall 2 is fucky
4. If in the future we want to do log uploading and such, it would be much better to do that from a separate process rather than from the game process

Reply:
We can remove it later? Installers should clean up after themselves anyways and even if not, having a floating exe in the install dir breaks nothing

![grrr](https://cdn.discordapp.com/emojis/971350699122253877.png)

Reply:
> We can remove it later? Installers should clean up after themselves anyways [...]

Installers outside of FlightCore are slow to update (VTOL still doesn't support packages for example) and unfortunately 1/4 of Northstar is still on manual installs x_x

> [...]  and even if not, having a floating exe in the install dir breaks nothing

Also true but not super nice to litter folder with "random" stuff

Reply:
At the very least I'd say we should make sure that by just looking at the exe it's clear it comes from Northstar, i.e. rename the binary from the current `CrashMsg.exe` to something like `NorthstarCrashMsg.exe`.

Also, is there a specific reason it's placed in `bin/`? Could we also stick it in profile dir like `Northstar.dll` or is there a reason preventing that?

Reply:
> At the very least I'd say we should make sure that by just looking at the exe it's clear it comes from Northstar, i.e. rename the binary from the current `CrashMsg.exe` to something like `NorthstarCrashMsg.exe`.

Stubs don't do this and i dont think this is necessary 

> Also, is there a specific reason it's placed in `bin/`? Could we also stick it in profile dir like `Northstar.dll` or is there a reason preventing that?

r5reloaded also has a `crashmsg.exe` in `bin/`. I don't see any reason to put it into profile dir

Reply:
> > At the very least I'd say we should make sure that by just looking at the exe it's clear it comes from Northstar, i.e. rename the binary from the current `CrashMsg.exe` to something like `NorthstarCrashMsg.exe`.
> 
> Stubs don't do this and i dont think this is necessary

Good point, paging @pg9182 for this (if he responds in time :>)

Reply:
On that note, I assume there isn't an easy way to keep logic inside currently existing binaries and have the move to separate binary be a separate PR, right?

Cause as far as I can tell, this is the only thing blocking so splitting it up would help fast tracking this PR here ^^"

Reply:
> On that note, I assume there isn't an easy way to keep logic inside currently existing binaries and have the move to separate binary be a separate PR, right?

This is really easy and testing locally apart from the game window staying open it seems to behave correctly ( almost as if i wrote this to prevent duplicate messages :> ). If this is really blocking i can push this change rn

Reply:
> If this is really blocking i can push this change rn

It *really* shouldn't be blocking

Reply:
> > If this is really blocking i can push this change rn
> 
> It _really_ shouldn't be blocking

This is the first time we are adding a new binary file since stubs and the first time we are adding an additional binary from Launcher code since wsock32 proxy both of which are over a year ago. Usually I'd poke @BobTheBob9 or @pg9182 for this but they'd take forever to respond ._.

Hence it would probably be best to just have integrated for now and make separate PR for separating out. That way we can have the rewrite merged now as opposed to getting potentially stuck in PR hell.

As I said, we don't have an easy way of removing files again (especially outside R2Northstar folder) so I'm always double/triple checking when we do add new ones. While I don't anticipate us having to remove the exe for some reason, I just really wanna make sure that we don't run into this case if we can avoid it ^^

Reply:
We *dont* want to avoid adding a new file, we *want* a new exe. Otherwise you're stuck with a stack of crash messages from audio overrides.
Having an additional exe is really no problem.

Reply:
I'm really not a fan of adding another binary and think the surface area that the user can interact with should be minimal.

Reply:
Do you honest to god think any user executes a random exe in the `bin/` directory that's called `CrashMsg.exe`

Reply:
Yes, but that is not my point.

What happens when the CrashMsg app is missing or broken or intercepted or replaced?
It adds another point of failure that we cannot manage.

Reply:
The same thing that happens when someone deletes the NorthstarLauncher.exe, Northstar.dll, wsock proxy or any other binary from the game.
No app randomly deletes or modifies binaries placed in a bin directory so this is really not an issue

Reply:
If it's missing the crash log still gets printed, just the message box doesn't appear.

r5reloaded has been running the same setup for god knows how long and they never had any issues

Reply:
r5reloaded also requires you to have a specific game version and download a 1.5GB zip.

Its not exactly a sane comparison to make.

Reply:
> It adds another point of failure that we cannot manage.

It does also remove some points of failure though. We have had issues in the past where the crash handler crashed, as well as times where the crash handler didnt catch things properly

Also, the r5r comparison is a good one to make, given that a large chunk of our code is adapted from r5r

Reply:
> Do you honest to god think any user executes a random exe in the `bin/` directory that's called `CrashMsg.exe`

You several underestimate the skill of some users to fuck things up :trollface: 





> > On that note, I assume there isn't an easy way to keep logic inside currently existing binaries and have the move to separate binary be a separate PR, right?
> 
> This is really easy and testing locally apart from the game window staying open it seems to behave correctly ( almost as if i wrote this to prevent duplicate messages :> ). If this is really blocking i can push this change rn

Honestly the best solution IMO. Split up the rewrite and moving to separate exe into separate PRs and then we can merge what is fine rn already ^^

Reply:
> You several underestimate the skill of some users to fuck things up :trollface:

If they install CrashMsg.exe wrong, they installed northstar wrong, not caused by this PR. If an installer messes this up it's not caused by this PR too.

> Honestly the best solution IMO. Split up the rewrite and moving to separate exe into separate PRs and then we can merge what is fine rn already ^^

I suppose this is the best solution for this

Reply:
Removed the extra exe, will make a separate PR with it

Reply:
> Removed the extra exe, will make a separate PR with it

Shall this PR be closed then?

Reply:
> Shall this PR be closed then?

No this is the PR without the separate exe, i.e. the one we wanna merge :P